### PR TITLE
WIP: Deneb protos separated from deneb-integration

### PR DESCRIPTION
**Pull request by kasey**
**What type of PR is this?**
Other

**What does this PR do? Why is it needed?**
This PR separates `deneb-integration` protobuf changes to reduce the line count in the diff to develop. 

**Other notes for review**

DO NOT MERGE

The plan for this branch is to provide a reference point for a more reviewable deneb-integration diff - a 3rd branch `deneb-integration-remaining` is on the way, which will be set up in a PR against this branch (`deneb-integration-proto`). We can use that PR to get a cleaner diff (no proto changes). If `deneb-integration-remaining` diffs cleanly against `deneb-integration`, we can confidently fast-forward merge deneb-integration knowing that it has had this extra round of final review.

This will break attribution/history and likely cause a messy rebase. This is a second attempt after trying to also integrate consensus-types and gateway changes, which spiraled out into other areas of the code.

Files from the `proto` package that were left behind to avoid breaking changes when the rest of the branch is missing:

```
proto/eth/service/beacon_chain_service.pb.go
proto/eth/service/beacon_chain_service.pb.gw.go
proto/eth/service/beacon_chain_service.proto
```

### CODE REVIEW

File: proto/engine/v1/json_marshal_unmarshal.go [hunk 1]
```diff
@@ -499,11 +569,31 @@ func (p *PayloadAttributes) MarshalJSON() ([]byte, error) {
 
 // MarshalJSON --
 func (p *PayloadAttributesV2) MarshalJSON() ([]byte, error) {
+	withdrawals := p.Withdrawals
+	if withdrawals == nil {
```
Comment (nisdas) [inline proto/engine/v1/json_marshal_unmarshal.go:573]: why was this added in ?

File: proto/eth/v2/BUILD.bazel [hunk 2]
```diff
@@ -45,7 +44,18 @@ ssz_gen_marshal(
         "SignedBlindedBeaconBlockBellatrix",
         "SignedBeaconBlockCapella",
         "SignedBlindedBeaconBlockCapella",
+        "SignedBeaconBlockDeneb",
+        "SignedBlindedBeaconBlockDeneb",
+        "BlsToExecutionChange",
```
Comment (nisdas) [inline proto/eth/v2/BUILD.bazel:49]: ~~Why are we add this in deneb ?~~

File: proto/engine/v1/json_marshal_unmarshal.go (hunk 1 above)
Comment (kasey) [reply proto/engine/v1/json_marshal_unmarshal.go:573]: presumably this fixes an issue where an empty withdrawals list results in a nil slice. You're probably wondering why this isn't needed in capella/develop though, cc @terencechain who is on the blame.
